### DISCUSSION TITLE
Protect API with JWT Access Token generated by Supabase

### DISCUSSION THREAD

**Discussion by AhmedJeylani** [supabase/supabase#1424] [Questions]
Hi All!

I do apologise if this question is silly. I am building an API in .Net core and I was able to get most of it working.
However, I would like to protect the API by using JWT Access token generated when you signin to supabase.

I am not sure if this is possible.

would be good to know.

**Comment by steve-chavez** [comment]
> I am building an API in .Net core 
> I would like to protect the API by using JWT Access token

Hm. Supabase has a builtin API that's already protected by JWT. Plus it enables many other feature/integrations.

I suggest looking at the docs: https://supabase.io/docs/learn/auth-deep-dive/auth-deep-dive-jwts

**Reply by AhmedJeylani** [reply]
I see what you mean. I was able to use the JWT secret to protect the endpoint. It is because we may need to plug in multiple services and Supabase. 
In addition to this I was who the valid issuer would be in this context:

![Screenshot 2021-05-08 at 16 24 22](https://user-images.githubusercontent.com/20175462/117544538-e81fc800-b019-11eb-89b8-a7ceb364059b.png)

**Reply by steve-chavez** [reply]
There's no `iss` claim(issuer) on the JWT by default. gotrue is what produces the JWTs in Supabase(see [here](https://github.com/supabase/gotrue#json-web-tokens-jwt)), I suggest to raise an issue there on how to add the `iss` claim.
